In [1]:
# import all the filenames
from allfunctions import *

1  def LassoCVModel(filename):

2  def OMPCVModel(filename):

3  def GradientBoostingCVModel(filename):

4  def RandomForestCVModel(filename):

5  def RidgeCVModel(filename):

6  def ElasticNetCVModel(filename):

7  def SVRPolyCVModel(filename):

8  def SVRSigmoidCVModel(filename):

9  def SVRLinearCVModel(filename):

10 def SVRRbfCVModel(filename):

In [2]:
def wrapper():
    if __name__ == '__main__':
        global model_selection
        global functionList
        global filenames
        
        functionList = [ LassoCVModel, GradientBoostingCVModel, 
                 RandomForestCVModel, RidgeCVModel, ElasticNetCVModel, 
                 SVRPolyCVModel, SVRSigmoidCVModel, SVRLinearCVModel, 
                 SVRRbfCVModel ] #OMPCVModel

        filenames = ['ABC.pickle', 'BCD.pickle', 'CDA.pickle', 'DAB.pickle' ]
        
        model_selection = {}
        
        for f in functionList:
            print()
            for filename in filenames:
                out = f(filename)
                newScore = out[filename]['train_rmse_cv_4_picking']

                if filename in model_selection:
                    #compare scores and select more accurate model
                    oldScore = model_selection[filename]['train_rmse_cv_4_picking']
                    if(newScore < oldScore):
                        print("【【【】】】switching models for %s..." % filename)
                        print("【【【】】】newScore: %f < oldScore: %f" % (newScore, oldScore))
                        print("【【【】】】newModel: %s \n" % f)
                        print(100*'_')
                        model_selection[filename] = out[filename]
                else:
                    #no score to compare
                    print("【【】】adding new model for %s \n" % filename)
                    print(100*'_')
                    model_selection[filename] = out[filename]
    return model_selection

In [3]:
t = wrapper()


Dataset size read: train 224 and test 97 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 20.691380811147901}
Lasso(alpha=20.691380811147901, copy_X=True, fit_intercept=True,
   max_iter=10000, normalize=False, positive=False, precompute=False,
   random_state=None, selection='random', tol=0.0001, warm_start=False)

【【】】adding new model for ABC.pickle 

____________________________________________________________________________________________________
Dataset size read: train 289 and test 32 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 6.1584821106602545}
Lasso(alpha=6.1584821106602545, copy_X=True, fit_intercept=True,
   max_iter=10000, normalize=False, positive=False, precompute=False,
   random_state=None, selection='random', tol=0.0001, warm_start=False)

【【】】adding new model for BCD.pickle 

____________________________

/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)



Best parameters set found on development set:

{'alpha': 20.691380811147901}
Lasso(alpha=20.691380811147901, copy_X=True, fit_intercept=True,
   max_iter=10000, normalize=False, positive=False, precompute=False,
   random_state=None, selection='random', tol=0.0001, warm_start=False)

【【】】adding new model for DAB.pickle 

____________________________________________________________________________________________________

Dataset size read: train 224 and test 97 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'learning_rate': 0.001, 'loss': 'ls'}
GradientBoostingRegressor(alpha=0.9, criterion='friedman_mse', init=None,
             learning_rate=0.001, loss='ls', max_depth=None,
             max_features=None, max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=1,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, presort='auto', random_state=None

In [4]:
t

{'ABC.pickle': {'model': RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
             max_features='auto', max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=2,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, n_jobs=-1, oob_score=True, random_state=None,
             verbose=0, warm_start=False),
  'test_mean_y_4_comparing': 1274.7285567010308,
  'test_rmse_4_reporting': 1071.6143340338772,
  'train_rmse_cv_4_picking': 487.31095240973497},
 'BCD.pickle': {'model': RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
             max_features='auto', max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=1,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, n_jobs=-1, oob_score=True, random_state=None,
             verbose=0, warm_start=False),
  'test_mean_y_4_comparing': 108.5103125,
  'test_rmse_4_reporti

In [5]:
import pickle

with open('model_selection_4C3.pickle', 'wb') as handle:
    pickle.dump(t, handle, protocol=pickle.HIGHEST_PROTOCOL)

In [7]:
# (4C3) Response when three of four modules are taken as training set
for key in t.keys():
    print(key)
    print(t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

ABC.pickle
487.31095241 1071.61433403 1274.7285567
____________________________________________________________________________________________________
BCD.pickle
431.548598884 1351.37874029 108.5103125
____________________________________________________________________________________________________
CDA.pickle
119.263406625 2201.7713798 2371.57863158
____________________________________________________________________________________________________
DAB.pickle
496.061206434 1065.77685249 1663.21
____________________________________________________________________________________________________


'''
# (4C3) Response when three of four modules are taken as training set

ABC.pickle
487.31095241 1071.61433403 1274.7285567
____________________________________________________________________________________________________
BCD.pickle
431.548598884 1351.37874029 108.5103125
____________________________________________________________________________________________________
CDA.pickle
119.263406625 2201.7713798 2371.57863158
____________________________________________________________________________________________________
DAB.pickle
496.061206434 1065.77685249 1663.21
____________________________________________________________________________________________________
'''

# (4C2) Response when two of four modules were taken as training set
'''  
for key in t.keys():
    print(key)
    print(t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')
    
AB.pickle
613.228034812 987.575604969 1468.96927835
____________________________________________________________________________________________________
BC.pickle
508.487190501 1529.27220935 985.434108527
____________________________________________________________________________________________________
CD.pickle
139.115733908 2032.01164137 1801.35669291
____________________________________________________________________________________________________
AC.pickle
139.211515629 1671.60046738 1817.44083333
____________________________________________________________________________________________________
AD.pickle
85.9832652301 2228.17276351 2371.57863158
____________________________________________________________________________________________________
BD.pickle
490.776589487 1094.31527963 1277.54806202
____________________________________________________________________________________________________

''' 